### DISCUSSION TITLE
CONTRIBUTING.md

### DISCUSSION THREAD

**Discussion by mcatta** [androiddevelopersitalia/aditogether#50] [Ideas]
Given the project is public and developers are going to work on it, I think it could be a good idea to create a `CONTRIBUTING.md` in order to provide our conventions in terms of:
- branch/commits convention naming
- linting rules (or at least put a link about the rules file)
- explain the purpose of the `tuner.sh`

Maybe @4face-studi0 @fondesa have other ideas, because part of these things was discussed and put in place during the infra-step.

**Comment by fardavide** [comment]
The little over-engineer that's in me, suggests that we could automatize all the checks 😁

We should add a contributions page, but I'll keep it minimal, because reading is boring, when you have a new project to start playing with 🙂

I would add a few details about the tuner and links to readme sections where we add more details about the architecture & co. The rest to be done by pre-commit hooks and Danger on CI.

What do you think?

**Comment by mcatta** [comment]
I agree, I wouldn't create a long white paper, but at least with the minimum information about:
- how to set up the project
- what is the branch and commits convention

The purpose is to speed up the developer and avoid wasting time reading the pre-commit or CI outputs, I'm writing this because I have seen #49 , where parts of the feedback are related to a set of conventions that we took during the first discussions. Of course, they are defined in the pre-commit script, but listing them somewhere could help IMO.

**Reply by fardavide** [reply]
This is also true! We don't want to create a trial-and-error environment :)

**Comment by fondesa** [comment]
Ideally, the tuner scripts would be completely optional since everything would be managed by the CI and they would be just additional checks to have a faster local failure.

Said that, I agree that having a `CONTRIBUTING.md` is surely beneficial
